cmd/tools/gendynamicconfig/dynamic_config.tmpl

```diff
@@ -0,0 +1,167 @@
+{{- /*gotype: go.temporal.io/server/cmd/tools/gendynamicconfig.dynamicConfigData*/ -}}
+// Code generated by cmd/tools/gendynamicconfig. DO NOT EDIT.
+
+package dynamicconfig
+
+import (
+	"time"
+
+	enumspb "go.temporal.io/api/enums/v1"
+	enumsspb "go.temporal.io/server/api/enums/v1"
+	"go.temporal.io/server/common/namespace"
+)
+
+const (
+	PrecedenceUnknown Precedence = iota
+{{- range .Precedences}}
+	Precedence{{.Name}}
+{{- end}}
+)
+{{$Precedences := .Precedences }}
+{{- range $T :=.Types}}
+	{{- range $P := $Precedences}}
+{{ if $T.IsGeneric -}}
+type {{$P.Name}}TypedSetting[T any] setting[T, func({{$P.GoArgs}})]
+
+// New{{$P.Name}}TypedSetting creates a setting that uses mapstructure to handle complex structured
```

Comment:
aw, this is going to conflict with 7052 that's almost ready. I'll handle the conflicts